File: daily/2025/2661_First_Completely_Painted_Row_or_Column.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int firstCompleteIndex(vector<int>& arr, vector<vector<int>>& mat) {
        int m = mat.size();
        int n = mat[0].size();
        vector<pair<int,int>> mapping(m * n + 1);
        for (int i = 0; i < m; i++){
            for (int j = 0; j < n; j++){
                mapping[mat[i][j]] = {i,j};
            }
        }
        unordered_map<int,int> row;
        unordered_map<int,int> col;
        for (int i = 0; i < m * n; i++){
            int r = mapping[arr[i]].first;
            int c = mapping[arr[i]].second;
            row[r]++;
            col[c]++;
            if (row[r] == n || col[c] == m) return i;
        }
        return m * n;
    }
};

int main(){
    vector<int> arr ={1,3,4,2};
    vector<vector<int>> mat = {{1,4},{2,3}};
    // vector<int> arr ={2,8,7,4,1,3,5,6,9};
    // vector<vector<int>> mat = {{3,2,5},{1,4,6},{8,7,9}};
    Solution S;
    cout << S.firstCompleteIndex(arr, mat) << endl;
    return 0;
}

File: daily_python/2661_First_Completely_Painted_Row_or_Column.py
import os
import sys
from typing import List

class Solution:
    def firstCompleteIndex(self, arr: List[int], mat: List[List[int]]) -> int:
        m = len(mat)
        n = len(mat[0])
        pos = [0] * (m * n + 1)
        for i in range(m):
            for j in range(n):
                pos[mat[i][j]] = (i,j)
        row = [0] * m
        col = [0] * n
        for i in range(m * n):
            r = pos[arr[i]][0]
            c = pos[arr[i]][1]
            row[r] += 1
            col[c] += 1
            if row[r] == n or col[c] == m:
                return i
        return m * n
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            arr = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            mat = [list(map(int, i.split(','))) for i in lines[1].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append((arr, mat))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for arr, mat in dataset:
        results.append(solution.firstCompleteIndex(arr, mat))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')